roles/run_hook/tasks/main.yml

```diff
@@ -18,7 +18,8 @@
   vars:
     _list_hooks: >-
       {{
-        hostvars[inventory_hostname][step] |
+        hostvars[inventory_hostname][step] is defined |
+        ternary(hostvars[inventory_hostname][step], hook_list) |
```

Comment:
I don't really get where that comes from, nor why you would inject hooks in yet another parameter?

You can also check how hooks are managed from within the architecture deployment, for instance:
https://github.com/openstack-k8s-operators/ci-framework/blob/5b499ca3867ac36eafd7de43d6a929ec0f71fd29/roles/kustomize_deploy/tasks/execute_step.yml#L104-L111

You can use `hooks: [list]` already.

Reply:
Done